### PR TITLE
Added support for architecture components lifecycle

### PR DESCRIPTION
**Pull request by kletzander**
Added functionality to cover issue https://github.com/yshrsmz/KeyboardVisibilityEvent/issues/48

### CODE REVIEW

File: keyboardvisibilityevent/build.gradle [hunk 1]
```diff
@@ -21,6 +21,7 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.50"
+    implementation "androidx.lifecycle:lifecycle-extensions:2.2.0"
```
Comment (yshrsmz) [inline keyboardvisibilityevent/build.gradle]: [lifecycle-extensions is deprecated.](https://developer.android.com/jetpack/androidx/releases/lifecycle#declaring_dependencies)

use `androidx.lifecycle:lifecycle-runtime` instead.

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt [hunk 2]
```diff
@@ -14,6 +18,23 @@ private const val KEYBOARD_MIN_HEIGHT_RATIO = 0.15
  */
 object KeyboardVisibilityEvent {
 
+    @JvmStatic
+    fun setEventListener(
```
Comment (yshrsmz) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt]: Could you add documentation for this method?

File: keyboardvisibilityevent/build.gradle (hunk 1 above)
Comment (kletzander) [reply keyboardvisibilityevent/build.gradle]: Replaced

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt (hunk 2 above)
Comment (kletzander) [reply keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt]: Documentation added

**Pull request merged**